lib/house.rb

```diff
@@ -0,0 +1,51 @@
+class House
+  attr_reader :price,
+              :address,
+              :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+    @total = 0
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    @price.delete("$").to_i > 500000
+  end
+
+  def rooms_from_category(room_category)
+    rooms.find_all do |room|
+      room.category == room_category
+    end
+  end
+
+  def area
```

Comment:
This method is close, but a bit flawed. It looks like we are adding to the instance variable `@total` every time this method is called. So if it is called more than twice our area would be twice the size, etc.

Reply:
I would create a local variable set to the value 0 here within this method if we want a fresh state the area calculation each time